rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -772,17 +788,17 @@ async fn run_controller(
                 let new_have_internet = network_changes::check_internet().context("Failed to check for internet")?;
                 if new_have_internet != have_internet {
                     have_internet = new_have_internet;
-                    if let Some(session) = controller.session.as_mut() {
+                    if controller.status.connlib_is_up() {
```

Comment:
Do we need to bother with this? Wouldn't it be simpler to always send the message and perhaps log something in the IPC client if connlib isn't signed in so it is a no-op?

Reply:
I was figuring it might save some tiny amount of battery on laptops if we don't send an IPC message that we know will be ignored. And we couldn't log it at `INFO` in the IPC service because then if the user is signed out it'll log every time anything changes

Reply:
Connlib has a log when it gets told to reset its state! (i.e. reconnect).

So we'd only need one on debug to say we are ignoring it if we aren't connected.